code/game/machinery/planet_beacon.dm

```diff
@@ -0,0 +1,216 @@
+#define BEACON_UNANCHORED 1
+#define BEACON_ANCHORED 2
+#define BEACON_IDLE 3
+#define BEACON_ACTIVE 4
+#define BEACON_DISTRESS 5
+
+#define DISTRESS_COOLDOWN 5 MINUTES // May need to be raised/lower in the future
+#define MAX_NAME_LENGTH 32 // How long of a name the planet can be renamed to
+
+/obj/machinery/power/planet_beacon
+	icon = 'icons/obj/machines/beacon.dmi'
+	icon_state = "beacon"
+	name = "EHF point beacon"
+	desc = "A long-range radio beacon for marking planetoids and points of interest for revisiting."
+	density = TRUE
+	anchored = FALSE
+	clicksound = "terminal_type"
+	use_power = NO_POWER_USE
+	idle_power_usage = IDLE_DRAW_MINIMAL
+	active_power_usage = ACTIVE_DRAW_MINIMAL
+	light_power = 1.5
+	var/state = BEACON_UNANCHORED
+	var/preserving = FALSE // Separate from state to prevent it being unpreserved on distress
+	var/obj/item/card/id/inserted_id
+	var/distress_time = 0 // value to compare with world.time if a distress beacon can be sent according to DISTRESS_COOLDOWN
+
+/obj/machinery/power/planet_beacon/Initialize()
+	. = ..()
+	AddComponent(/datum/component/gps, "EHF" + num2text(rand(0,9)))
+
+/obj/machinery/power/planet_beacon/process()
+	var/turf/T = get_turf(src)
+	var/obj/structure/cable/C = locate() in T
+	if(!C && powernet)
+		disconnect_from_network()
+		change_state(BEACON_ANCHORED)
+	if(state > BEACON_IDLE) // If active or sending distress, use active_power_usage
+		if(avail(active_power_usage))
+			add_load(active_power_usage)
+		else
+			change_state(BEACON_ANCHORED)
+	else if(state == BEACON_IDLE) // If idle, use idle_power_usage
+		if(avail(idle_power_usage))
+			add_load(idle_power_usage)
+		else
+			change_state(BEACON_ANCHORED)
+
+/obj/machinery/power/planet_beacon/attackby(obj/item/O, mob/user, params)
```

Comment:
Please do not use single letter vars - `obj/item/O`.

This goes for the whole PR.